lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,318 @@
+    onChange(displayName);
+    onSearch(displayName);
+    focusTextInput();
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  function handleKeyDown(evt: KeyboardEvent<HTMLElement>) {
+    const inputEle = evt.currentTarget as HTMLInputElement,
+        elToFocusText = focusableBtnIndex !== null ? matches[focusableBtnIndex].displayName : null;
+    switch (evt.key) {
+      case 'Enter':
+        if (focusableBtnIndex !== null) {
+          handleOnClick(matches[focusableBtnIndex]);
+        }
+        evt.preventDefault();
+        break;
+      case 'Home':
+        inputEle.setSelectionRange(0, 0);
+        evt.preventDefault();
+        break;
+      case 'End':
+        autoComplete && elToFocusText
+          ? inputEle.setSelectionRange(elToFocusText.length, elToFocusText.length)
+          : inputEle.setSelectionRange(value.length, value.length);
+        evt.preventDefault();
+        break;
+      case 'ArrowDown':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+
+        focusableBtnIndex !== null
+          ? (focusableBtnIndex === matches.length - 1 ? focusFirst() : focusNext())
+          : focusFirst();
+
+        evt.preventDefault();
+        break;
+      case 'ArrowUp':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+
+        focusableBtnIndex !== null
+          ? (focusableBtnIndex === 0 ? focusLast() : focusPrev())
+          : focusLast();
+
+        evt.preventDefault();
+        break;
+      case 'Backspace':
+        if (autoComplete) {
+          setInlineStyle(false);
+        }
+        break;
+      case 'Escape':
+        if (showDropdown) {
+          setShowDropdown(false);
+        }
+        else {
+          onChange('');
+          setFocusableBtnIndex(null);
+        }
+        break;
+      default:
+        if (autoComplete) {
+          setInlineStyle(true);
+        }
+        break;
+    }
+  }
+
+  useEffect(function() {
+    // Nullify focusableBtnIndex whenever the number of matches changes
+    setFocusableBtnIndex(null);
+
+    if (matches.length && autoComplete && showDropdown && inlineStyle) {
+      // If the typed characters match the beginning of the name of option in the dropdown,
+      // the first suggestion is automatically selected.
+      const firstOptVal = matches[0].displayName;
+      if (firstOptVal.toLowerCase().indexOf(value.toLowerCase()) === 0) {
+        setFocusableBtnIndex(0);
+      }
+    }
+  }, [matches]);
+
+  // Highlight the portion of the selected suggestion that has not been typed by the user and display
+  // a completion string inline after the input cursor in the input box.
+  useEffect(function() {
+    if (inlineStyle && focusableBtnIndex === 0) {
+      const firstOptVal = matches[0].displayName;
+      inputRef.current?.querySelector('input')?.setSelectionRange(value.length, firstOptVal.length);
+    }
+  }, [focusableBtnIndex]);
```

Comment:
Seems like `matches`, `inlineStyle`, and `value` should be in this dependency array too.

Reply:
Updated here 7047f0760fdf092735b418b118ce8b2643a1d17c.